apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs

```diff
@@ -627,7 +627,42 @@ internal DatabaseStatistics GetStatisticsSnapshot()
                         Healthy,
                         Shutdown);
                 }
-            }            
+            }
+
+            /// <summary>
+            /// Waits for the session pool to be populated up to its minimum size.
+            /// </summary>
+            /// <remarks>
+            /// If the pool is unhealthy or becomes unhealthy before it reaches its minimum size,
+            /// the returned task will be faulted with an <see cref="RpcException"/>.
+            /// </remarks>
+            /// <param name="cancellationToken">An optional token for canceling the call.</param>
+            /// <returns>A task which will complete when the session pool has reached its minimum size.</returns>
+            internal async Task WaitForPoolAsync(CancellationToken cancellationToken = default)
```

Comment:
The intention was that it didn't start anything itself - it's a "watching" method rather than a "doing" method. In reality, the tasks will be kicked off on construction (for all but the shutdown case) - in the next PR :) So starting extra tasks here wouldn't be necessary, and would probably make the tests a bit more confusing.